relnotes/taskpool.feature.md

```diff
@@ -0,0 +1,4 @@
+* `ocean.task.TaskPool`
+
+  A new function `awaitAllTasks()` has been added. This function suspends the
+  current task until all tasks in the pool have finished executing.
```

Comment:
I'll update the release notes and function documentation as you suggested.

>I also wonder about asserting that this is the case, but I'm not sure if there's a way to do that without iterating over the whole pool, which seems a bit excessive.

We're iterating over all the busy tasks in the pool anyway, so we'd just need to extend that to iterate over all tasks in the pool. The additional safety that the assertion would provide would make it worthwhile imo.

The tricky part when iterating over all tasks is to identify whether a task is busy or not. For this, I'll need to add a new `isBusy()` function to the API, which would mean touching a couple of other modules.

So for now, I'd say we add this `awaitRunningTasks()` function with just the note in the documentation, and I'll submit another PR to add the assertion after this is merged. How does that sound?